### PR TITLE
makes blockHash optional in chain/getTransaction requests

### PR DESCRIPTION
**Pull request by hughy**
## Summary

after adding an index from transaction hash to block hash in #3761 we no longer need to include a block hash in requests to chain/getTransaction. instead we can use the index to look up the hash of the block on the main chain that includes that transaction, if any.

- defines 'getHashByTransactionHash' to look up a block hash in 'transactionHashToBlockHash'
- changes 'blockHash' to an optional request parameter

## Testing Plan

- adds route tests for chain/getTransaction

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[X] Yes
```
https://github.com/iron-fish/website/pull/359

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/blockchain/blockchain.ts [hunk 1]
```diff
@@ -1049,6 +1049,14 @@ export class Blockchain {
     return hash || null
   }
 
+  async getHashByTransactionHash(
```
Comment (EvanJRichard) [inline ironfish/src/blockchain/blockchain.ts:1052]: is `getBlockHashByTransactionHash` too wordy, or unnecessary given where the function lives and its return type? I ask because when I initially read this, until I saw the return type, I was confused, "wait, if we have the hash why do we need a hash, is this a no-op?"

Comment (hughy) [reply ironfish/src/blockchain/blockchain.ts:1052]: I think that `getBlockHashByTransactionHash` would be clearer

I left out the `Block` because there are a lot of other methods like `getHashAtSequence` that have names that leave the 'block' part implicit. Since there are two different hashes at play here I think it would be better to specify

**Review by EvanJRichard — APPROVED** [review]

**Review by rohanjadvani — APPROVED** [review]
